### DISCUSSION TITLE
Audit fails in GitHub Actions … why?

### DISCUSSION THREAD

**Discussion by slhck** [Homebrew/discussions#4282] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.4-21-g6db7732
ORIGIN: https://github.com/Homebrew/brew
HEAD: 6db7732fa33ab808e405f8ac7673735edd2c8787
Last commit: 8 hours ago
Core tap: N/A
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_BOOTSNAP: set
HOMEBREW_CASK_OPTS: []
HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS: 3650
HOMEBREW_COLOR: set
HOMEBREW_CURL_PATH: /usr/bin/curl
HOMEBREW_DEVELOPER: set
HOMEBREW_FAIL_LOG_LINES: 150
HOMEBREW_GIT_EMAIL: 1589480+BrewTestBot@users.noreply.github.com
HOMEBREW_GIT_NAME: BrewTestBot
HOMEBREW_GIT_PATH: /usr/bin/git
HOMEBREW_MAKE_JOBS: 2
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_EMOJI: set
HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: set
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: dual-core 64-bit broadwell
Clang: 14.0.0
Git: 2.39.2 => /usr/bin/git
Curl: 7.81.0 => /usr/bin/curl
Kernel: Linux 5.15.0-1033-azure x86_64 GNU/Linux
OS: Ubuntu 22.04.2 LTS (jammy)
Host glibc: 2.35
/usr/bin/gcc: 11.3.0
/usr/bin/ruby: 3.0.2
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
empty
```

### Description of issue

We run CI on https://github.com/homebrew-ffmpeg/.

Recently our PR jobs failed:

https://github.com/homebrew-ffmpeg/homebrew-ffmpeg/actions/runs/4290180691/jobs/7474848675

The job fails with:

>   Error: No such file or directory @ realpath_rec - /home/runner/work/homebrew-ffmpeg/homebrew-ffmpeg/ffmpeg

Previous runs have always worked. Re-running did not help.

Any idea what might cause this?

**Comment by Bo98** [comment, marked as answer]
I don't think test-bot supports path-based arguments very well. It assumes you are using Homebrew/core (because it doesn't know otherwise - `ffmpeg.rb` could belong to anything) and that's exactly what it's trying to do - check ffmpeg in Homebrew/core. It possibly has always been doing this and only errored now because Homebrew/core source files aren't downloaded by default anymore since 4.0.0 and so can't check Homebrew/core's version of ffmpeg like it previously could.

test-bot does automatic detection of modified formulae if passed no arguments, but if that is not good enough I recommend passing `homebrew-ffmpeg/ffmpeg/ffmpeg` instead.

**Reply by slhck** [reply]
That seems to have worked, thanks!

https://github.com/homebrew-ffmpeg/homebrew-ffmpeg/actions/runs/4301068938/jobs/7497886072

**Reply by slhck** [reply]
That said, it fails when running CI on master: https://github.com/homebrew-ffmpeg/homebrew-ffmpeg/actions/runs/4301363571/jobs/7498504434

Apparently because there are no changes to test. So it might need arguments. (But then again, we don't have changes on master only, so testing from within PRs is fine.)
